ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveJoinPushTransitivePredicatesRule.java

```diff
@@ -74,6 +74,8 @@
   public static final HiveJoinPushTransitivePredicatesRule INSTANCE_ANTIJOIN =
           new HiveJoinPushTransitivePredicatesRule(HiveAntiJoin.class, HiveRelFactories.HIVE_FILTER_FACTORY);
 
+  private static final UnsafeOperatorsFinder UNSAFE_OPERATORS_FINDER = new UnsafeOperatorsFinder(true);
```

Comment:
I think we may end-up calling this visitor from multiple threads so I would either remove static or make the class immutable.